main.cpp

```diff
@@ -99,14 +100,127 @@ inf
 
 good to go!
 
+*/
 
+/*
+ MAKE SURE YOU ARE NOT ON THE MASTER BRANCH
 
+ Commit your changes by clicking on the Source Control panel on the left, entering a message, and click [Commit and push].
+ 
+ If you didn't already: 
+    Make a pull request after you make your first commit
+    pin the pull request link and this repl.it link to our DM thread in a single message.
 
+ send me a DM to review your pull request when the project is ready for review.
 
-*/
+ Wait for my code review.
+ */
 
 #include <iostream>
 
+struct FloatType
+{
+    float* value;
+    FloatType(float v) : value(new float(v)) {}
+
+    FloatType& add(float a)
+    {
+        *value += a;
+        return *this;
+    }
+
+    FloatType subtract(float a)
+    {
+        *value -= a;
+        return *this;
+    }
+
+    FloatType multiply(float a)
+    {
+        *value *= a;
+        return *this;
+    }
+
+    FloatType divide(float a)
+    {
+        if (a == .0f  || *value == .0f)
+        {
+            std::cout << "warning: floating point division by zero!\n";
+        }
+        *value /= a;
+        return *this;
+    }
+};
+
+struct DoubleType
+{
+    double* value;
+    DoubleType(double v) : value(new double(v)) {}
+
+    DoubleType& add(double a)
+    {
+        *value += a;
+        return *this;
+    }
+
+    DoubleType subtract(double a)
+    {
+        *value -= a;
+        return *this;
+    }
+
+    DoubleType multiply(double a)
+    {
+        *value *= a;
+        return *this;
+    }
+
+    DoubleType divide(double a)
+    {
+        if (a == .0 || *value == 0.0)
+        {
+            std::cout << "warning: floating point division by zero!\n";
+        }
+        *value /= a;
+        return *this;
+    }
+};
+
+struct IntType
+{
+    int* value;
+    IntType(int v) : value(new int(v)) {}
+
+    IntType& add(int a)
+    {
+        *value += a;
+        return *this;
+    }
+
+    IntType subtract(int a)
+    {
+        *value -= a;
+        return *this;
+    }
+
+    IntType multiply(int a)
+    {
+        *value *= a;
+        return *this;
+    }
+
+    IntType divide(int a)
+    {
+        if (a == .0 || value == nullptr)
```

Comment:
value can never be a nullptr because of how the class is constructed
This nullptr check is unnecessary
- [ ] revise

Reply:
Solved